### PR TITLE
Metric aggregation

### PR DESCRIPTION
**Pull request by Dokholyan**
## Before submitting (checklist)

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [ ] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [ ] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [ ] Did you check the docs with `make check-docs`?
- [ ] Did you write any new necessary tests?
- [ ] Did you check that your code passes the unit tests `pytest .` ? 
- [ ] Did you add your new functionality to the docs?
- [ ] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

PS
- [x] I know, that I could [join slack](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for pull request discussion.

### CODE REVIEW

File: catalyst/callbacks/aggregation.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+
+class MetricAggregationCallback(Callback):
+    """A callback to aggregate several metrics in one value."""
+
+    def __init__(
+        self,
+        prefix: str,
+        metrics: Union[str, List[str], Dict[str, float]] = None,
+        mode: str = "mean",
```
Comment (Scitator) [inline catalyst/callbacks/aggregation.py]: `mode: Union[str, Callable]`?

File: catalyst/callbacks/aggregation.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                    "the metrics must be specified "
+                    "and must be a dict"
+                )
+        elif not callable(mode):
+            raise NotImplementedError(
+                "mode must be `sum`, `mean` "
+                "or `weighted_sum` or `weighted_mean` or be Callable"
+            )
+
+        assert scope in ("batch", "loader", "epoch")
```
Comment (Scitator) [inline catalyst/callbacks/aggregation.py]: there is no `epoch` case handler, should we remove it?

File: catalyst/callbacks/aggregation.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            mode: function for aggregation.
+                Must be either ``sum``, ``mean`` or ``weighted_sum`` or user's
+                function to aggregate metrics. This function must get dict of
+                metrics and runner and return aggregated metric. It can be
+                useful for complicated fine tuning with different losses that
+                depends on epochs and loader or something also
+            scope: type of metric. Must be either ``batch``, ``loader`` or
+                ``epoch``
+            multiplier: scale factor for the aggregated metric.
+        Examples:
```
Comment (Scitator) [inline catalyst/callbacks/aggregation.py:37]: could you please also add a fully working example? something like:
```python
import torch
from torch.utils.data import DataLoader, TensorDataset
from catalyst import dl

# sample data
num_samples, num_features, num_classes = int(1e4), int(1e1), 4
X = torch.rand(num_samples, num_features)
y = (torch.rand(num_samples, ) * num_classes).to(torch.int64)

# pytorch loaders
dataset = TensorDataset(X, y)
loader = DataLoader(dataset, batch_size=32, num_workers=1)
loaders = {"train": loader, "valid": loader}

# model, criterion, optimizer, scheduler
model = torch.nn.Linear(num_features, num_classes)
criterion = {"ce": torch.nn.CrossEntropyLoss(), "bce": torch.nn.BCEWithLogitsLoss()}
optimizer = torch.optim.Adam(model.parameters())
scheduler = torch.optim.lr_scheduler.MultiStepLR(optimizer, [2])

class CustomSupervisedRunner(dl.Runner):
    def handle_batch(self, batch):
        x, y = batch
        logits = self.model(x)
        num_classes = logits.shape[-1]
        self.batch = {
            "features": x,
            "logits": logits,
            "targets": y,
            "targets_onehot": torch.nn.functional.one_hot(y, num_classes=num_classes).float()
        }

# model training
runner = CustomSupervisedRunner()
runner.train(
    model=model,
    criterion=criterion,
    optimizer=optimizer,
    scheduler=scheduler,
    loaders=loaders,
    logdir="./logdir",
    num_epochs=3,
    callbacks=[
        dl.AccuracyCallback(input_key="logits", target_key="targets", num_classes=num_classes),
        dl.CriterionCallback(input_key="logits", target_key="targets", metric_key="loss_ce", criterion_key="ce"),
        dl.CriterionCallback(input_key="logits", target_key="targets_onehot", metric_key="loss_bce", criterion_key="bce"),
        dl.OptimizerCallback(metric_key="loss_ce")
    ]
)
```

Comment (Scitator) [reply catalyst/callbacks/aggregation.py:37]: but with metric aggregation in the end

File: catalyst/callbacks/__init__.py [hunk 4]
```diff
@@ -23,6 +23,7 @@
     ILRUpdater,
     LRFinder,
 )
+from catalyst.callbacks.aggregation import MetricAggregationCallback
```
Comment (Scitator) [inline catalyst/callbacks/__init__.py:26]: btw, add this callback to the docs please, here - https://github.com/catalyst-team/catalyst/blob/dev/docs/api/callbacks.rst

File: catalyst/callbacks/aggregation.py (hunk 1 above)
Comment (Dokholyan) [reply catalyst/callbacks/aggregation.py]: You are right

File: catalyst/callbacks/aggregation.py (hunk 2 above)
Comment (Dokholyan) [reply catalyst/callbacks/aggregation.py]: It seems to us that we do not need it. Since epoch_metrics is just copied from loader_metrics

File: docs/api/callbacks.rst [hunk 5]
```diff
@@ -57,6 +57,11 @@ Metric – LoaderMetricCallback
     :exclude-members: on_experiment_start, on_stage_start, on_epoch_start, on_loader_start, on_batch_start, on_batch_end, on_loader_end, on_epoch_end, on_stage_end, on_experiment_end
     :show-inheritance:
 
+.. autoclass:: catalyst.callbacks.aggregation.MetricAggregationCallback
+    :members:
+    :exclude-members: on_batch_end, on_loader_end
```
Comment (Scitator) [inline docs/api/callbacks.rst]:
```suggestion
    :exclude-members: on_experiment_start, on_stage_start, on_epoch_start, on_loader_start, on_batch_start, on_batch_end, on_loader_end, on_epoch_end, on_stage_end, on_experiment_end
```

File: catalyst/callbacks/aggregation.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
+from typing import Any, Callable, Dict, List, Union
+
+import torch
+
+from catalyst.dl import Callback, CallbackNode, CallbackOrder
```
Comment (Scitator) [inline catalyst/callbacks/aggregation.py]:
```suggestion
from catalyst.core.callback import Callback, CallbackNode, CallbackOrder
```

File: catalyst/callbacks/aggregation.py (hunk 2 above)
Comment (Scitator) [reply catalyst/callbacks/aggregation.py]: let's remove than
